keps/1432-admission-check-per-resource-flavor/README.md

```diff
@@ -0,0 +1,288 @@
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a user who has reserved machines at my cloud provider I would like to use them first. If they are not obtainable then
+switch to spot machines, using ProvisioningRequest. It only makes sense to run Provisioning AdmissionCheck when the ResourceFlavor is Spot.
+
+
+
+### Notes/Constraints/Caveats (Optional)
+All `AdmissionChecks` assigned to a Workload must have different controllers.
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+We propose adding a new field to `FlavorQuota` API called `AdmissionChecks` in following manner:
+
+```
+type FlavorQuotas struct {
+  // name of this flavor. The name should match the .metadata.name of a
+  // ResourceFlavor. If a matching ResourceFlavor does not exist, the
+  // ClusterQueue will have an Active condition set to False.
+  Name ResourceFlavorReference `json:"name"`  
+
+  // resources is the list of quotas for this flavor per resource.
+  // There could be up to 16 resources.
+  // +listType=map
+  // +listMapKey=name
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=16
+  Resources []ResourceQuota `json:"resources"`
+
+  // admissionChecks is the list of AdmissionChecks that should run when the ResourceFlavor is used.
+  // No two AdmissionChecks should have the same controller.
+  // +listType=set
+  AdmissionChecks []string `json:"admissionchecks"`
+}
+```
+
+At the same time, we want to preserve the existing `AdmissionChecks` field in `ClusterQueue` API. A Workload may have
+assigned `AdmissionChecks` both from `ClusterQueue` and `FlavorQuota` APIs. In case two `AdmissionChecks` have the same
+controller, the one from `FlavorQuota` API should take over.
+
+This enables having default `AdmissionChecks` that apply to all Workloads submitted to a `ClusterQueue`, and more
+specific ones that apply only to Workloads using a specific `ResourceFlavor`.
```

Comment:
Do we have a good story for supporting both levels at the same time? If so please add to the stories section. Otherwise I would suggest to simplify and only allow them to be present at one level. We can relax this in the future if needed. 

It feels that setting admission checks at both levels might complicate debugging for the "why my workload is not starting?" question.

Reply:
I've added a story for supporting this. Maybe we could simplify the mechanism though and discard overriding AdmissionChecks. WDYT

Reply:
If the story is not yet motivated by a user request, then I would suggest to simplify. 

In case we have ACs only on one level we can apply the same validation rules, but scoped.

If we allow setting the Admissionchecks at two levels the semantic of merging and validation gets tricky. Some questions to emphasize the complications: 
1. why override, if we allow some admission checks (like ProvReq) to have multiple instances we could append rather than override,
2, why override rather than validation error for admission checks that require single instance, say for MK, user might get surprised which one "wins"
3. are the answers above different if the ACs have different / the same name?

Reply:
Thanks for clarifying. I took that into consideration and overall agree with it. I've committed changes that aligns with that approach